ci/build-src-tarball.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+set -ex
+
+# Determine the name of the tarball
+tag=dev
+if [[ $GITHUB_REF == refs/tags/v* ]]; then
+  tag=${GITHUB_REF:10}
+fi
+pkgname=wasmtime-$tag-src
+rm -rf /tmp/$pkgname
+mkdir /tmp/$pkgname
+
+# Vendor all crates.io dependencies since this is supposed to be an
+# offline-only-compatible tarball
+mkdir /tmp/$pkgname/.cargo
+cargo vendor > /tmp/$pkgname/.cargo/config.toml
+
+# Copy over everything in-tree to the destination
+cp -r * /tmp/$pkgname
+
+# Create the tarball from the destination
+mkdir -p dist
+tar -czf dist/$pkgname.tar.gz -C /tmp $pkgname
```

Comment:
I'm somewhat hesitant to do that because we haven't done it for any other tarballs and I'm always somewhat scared of using fancy `tar` options in fear that we'll eventually get a `tar` on some system which is too old

Reply:
Sure, that's fair. Though I just did a quick check out of curiosity, and `--sort` is the newest of these flags, and even the version of GNU tar in Ubuntu 16.04 had that option. Since this is only supposed to run on GitHub runners I guess I wouldn't be as concerned about whether random systems have the same options available.